.github/workflows/pr-build.yml

```diff
@@ -0,0 +1,98 @@
+name: "PR Build"
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    name: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - id: setup-java-8
+        name: Setup Java 8
+        uses: actions/setup-java@v2
+        with:
+          distribution: adopt
+          java-version: 8
+      - uses: burrunan/gradle-cache-action@v1
+        name: restore gradle cache
+        with:
+          gradle-version: wrapper
+          job-id: pr-build
+          remote-build-cache-proxy-enabled: false
+      - name: ./gradlew lint
+        run: ./gradlew lint
+      - uses: burrunan/gradle-cache-action@v1
+        name: ./gradlew assemble
+        with:
+          arguments: --stacktrace assemble
+          gradle-version: wrapper
+          job-id: java-contrib
+          remote-build-cache-proxy-enabled: false
+      - name: ./gradlew test
+        uses: burrunan/gradle-cache-action@v1
+        id: gradle-test
+        timeout-minutes: 5
+        with:
+          arguments: --stacktrace --max-workers 2 test
+          gradle-version: wrapper
+          job-id: java-contrib
+          remote-build-cache-proxy-enabled: false
```

Comment:
I'm don't know GitHub Actions well, so likely don't know what I'm talking about 😁, but you might be able to have just one `uses: burrunan/gradle-cache-action@v1`, and still have multiple commands after that, e.g.

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/f956a58c0abd44391a5df2bf54748cfa65277e07/.github/workflows/pr.yaml#L124-L130

Reply:
I thought so too, but didn't see `UP-TO-DATE` markers in the logs for repeated tasks without doing so.  I'm very new to gh actions so will keep looking into it.